ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CommonStorageBackend.scala

```diff
@@ -383,51 +383,50 @@ private[backend] trait CommonStorageBackend[DB_BATCH] extends StorageBackend[DB_
       .asVectorOf(partyEntryParser)(connection)
   }
 
-  private val SQL_SELECT_MULTIPLE_PARTIES =
-    SQL(
-      "select parties.party, parties.display_name, parties.ledger_offset, parties.explicit, parties.is_local from parties, parameters where party in ({parties}) and parties.ledger_offset <= parameters.ledger_end"
-    )
-
-  private case class ParsedPartyData(
-      party: String,
-      displayName: Option[String],
-      ledgerOffset: Offset,
-      explicit: Boolean,
-      isLocal: Boolean,
-  )
-
-  private val PartyDataParser: RowParser[ParsedPartyData] = {
-    import com.daml.platform.store.Conversions.columnToOffset
+  val partyDetailsParser: RowParser[PartyDetails] = {
     import com.daml.platform.store.Conversions.bigDecimalColumnToBoolean
-    Macro.parser[ParsedPartyData](
-      "party",
-      "display_name",
-      "ledger_offset",
-      "explicit",
-      "is_local",
-    )
+    str("party") ~
+      str("display_name").? ~
+      bool("is_local") map { case party ~ displayName ~ isLocal =>
+        PartyDetails(
+          party = Ref.Party.assertFromString(party),
+          displayName = displayName,
+          isLocal = isLocal,
+        )
+      }
   }
 
-  private def constructPartyDetails(data: ParsedPartyData): PartyDetails =
-    PartyDetails(Ref.Party.assertFromString(data.party), data.displayName, data.isLocal)
-
-  def parties(parties: Seq[Ref.Party])(connection: Connection): List[PartyDetails] = {
-    import com.daml.platform.store.Conversions.partyToStatement
-    SQL_SELECT_MULTIPLE_PARTIES
-      .on("parties" -> parties)
-      .as(PartyDataParser.*)(connection)
-      .map(constructPartyDetails)
+  private def queryParties(parties: Set[String], connection: Connection): Vector[PartyDetails] = {
+    val partyFilter =
+      if (parties.isEmpty) cSQL""
```

Comment:
What about using `Option[Set[String]]` instead of redefining the meaning of filtering for the empty set?

Reply:
I was thinking about that, but it seemed to me the empty set is anyhow an invalid case here. Since you mentioned, I am changing it.